include/functions/random.hpp

```diff
@@ -32,6 +32,9 @@ BigInt big_random(size_t num_digits = 0) {
 
     BigInt big_rand;
     big_rand.value = "";    // clear value to append digits
+
+    // ensure first digit is non-zero
+    big_rand.value += std::to_string(rand_generator() % 8 + 1);
```

Comment:
The expression should be be `1 + rand_generator() % 9`.

Reply:
Wow, I'm a bozo. I'll fix that.